homeassistant/components/hydrawise.py

```diff
@@ -0,0 +1,156 @@
+"""
+Support for Hydrawise cloud.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/hydrawise/
+"""
+import asyncio
+from datetime import timedelta
+import logging
+
+from requests.exceptions import ConnectTimeout, HTTPError
+import voluptuous as vol
+
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_ACCESS_TOKEN, CONF_SCAN_INTERVAL)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import (
+    async_dispatcher_connect, dispatcher_send)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+
+REQUIREMENTS = ['hydrawiser==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ALLOWED_WATERING_TIME = [5, 10, 15, 30, 45, 60]
+
+CONF_ATTRIBUTION = "Data provided by hydrawise.com"
+CONF_WATERING_TIME = 'watering_minutes'
+
+NOTIFICATION_ID = 'hydrawise_notification'
+NOTIFICATION_TITLE = 'Hydrawise Setup'
+
+DATA_HYDRAWISE = 'hydrawise'
+DOMAIN = 'hydrawise'
+DEFAULT_WATERING_TIME = 15
+
+DEVICE_MAP_INDEX = ['KEY_INDEX', 'ICON_INDEX', 'DEVICE_CLASS_INDEX',
+                    'UNIT_OF_MEASURE_INDEX']
+DEVICE_MAP = {
+    'auto_watering': ['Automatic Watering', 'mdi:autorenew', '', ''],
+    'is_watering': ['Watering', '', 'moisture', ''],
+    'manual_watering': ['Manual Watering', 'mdi:water-pump', '', ''],
+    'next_cycle': ['Next Cycle', 'mdi:calendar-clock', '', ''],
+    'status': ['Status', '', 'connectivity', ''],
+    'watering_time': ['Watering Time', 'mdi:water-pump', '', 'min'],
+    'rain_sensor': ['Rain Sensor', '', 'moisture', '']
+}
+
+BINARY_SENSORS = ['is_watering', 'status', 'rain_sensor']
+
+SENSORS = ['next_cycle', 'watering_time']
+
+SWITCHES = ['auto_watering', 'manual_watering']
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+SIGNAL_UPDATE_HYDRAWISE = "hydrawise_update"
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_ACCESS_TOKEN): cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL, default=SCAN_INTERVAL):
+            cv.time_period,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the Hunter Hydrawise component."""
+    conf = config[DOMAIN]
+    access_token = conf.get(CONF_ACCESS_TOKEN)
```

Comment:
Don't use `dict.get(key)` for required config items, use `dict[key]`.

Reply:
Fixed.